students/shibata/session3/mailroom.py

```diff
@@ -0,0 +1,150 @@
+#! /usr/bin/env Python
+"""
+It should have a data structure that holds a list of your donors and a history of the amounts they have donated. This structure should be populated at first with at least five donors, with between 1 and 3 donations each.
+
+You can store that data structure in the global namespace.
+The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”)
+"""
+
+from time import sleep
+import pprint
+
+if __name__ == '__main__':
+    objFileName = "C:\Files\donate.txt"
+    strData = ""
+    dicRow = {}
+    lstTable = []
+
+# Step 1
+# When the program starts, load the any data you have
+# in a text file called ToDo.txt into a python Dictionary.
+objFile = open(objFileName, "r")
+for line in objFile:
+    strData = line.split(",") # readline() reads a line of the data into 2 elements
+    dicRow = {"Name":strData[0].strip(), "Total":strData[1].strip(), "Num":strData[2].strip()}
+    lstTable.append(dicRow)
+objFile.close()
+
+# Display a menu of choices to the user
+while(True):
+    print ("""
+    Menu of Actions
+    1) Send a Thank You
+    2) Create a Report
+    3) quit
+    """)
+    strChoice = str(input("choose from a menu of 3 actions [1 to 3] - "))
+    print()#adding a new line
+
+    if (strChoice.strip() == '1'):
+        #   print("Donor Name                | Total Given | Num Gifts | Average Gift\n------------------------------------------------------------------")
+
+                while(True):
+                    strName = str(input("What is Donor Name? (type list or back for menu)- ")).strip()
+                    print()#adding a new line
+                    if (strName.strip() == 'list'):
+                        print("Donor Name | Total Given | Num Gifts | Average Gift\n------------------------------------------------------------------")
+                        #        for dicRow in lstTable:
+                        #            print(dicRow)
+                                        #4a Show the current items in the table
+                        for row in lstTable:
+                            print ("{}, ${:.2f}, {}, ${:.2f}".format(row["Name"], float(row["Total"]), row["Num"], float(row["Total"])/int(row["Num"])))
```

Comment:
you shouldnlt need to call float() -- the float string formatter can deal with integers.

Reply:
And yo umay not have leanred that yet, but you can use a dict with format strings:

print ("{Name}, ${Total:.2f}, {Num}, ${Avg:.2f}".format(**row))

(though you'd need to put Avg in the dict first.)